Storage/src/StorageClient.php

```diff
@@ -425,4 +416,125 @@ public function getServiceAccount(array $options = [])
+     *     'serviceAccountEmail' => $serviceAccountEmail
+     * ]);
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $serviceAccountEmail If present, only keys for the given
+     *           service account are returned.
+     *     @type bool $showDeletedKeys Whether or not to show keys in the
+     *           DELETED state.
+     *     @type string $userProject If set, this is the ID of the project which
+     *           will be billed for the request.
+     * }
+     * @return ItemIterator<HmacKey>
+     */
+    public function hmacKeys(array $options = [])
+    {
+        $this->requireProjectId();
+
+        $resultLimit = $this->pluck('resultLimit', $options, false);
+        return new ItemIterator(
+            new PageIterator(
+                function (array $metadata) {
+                    return $this->hmacKey(
+                        $metadata['accessId'],
+                        $metadata
+                    );
+                },
+                [$this->connection, 'listHmacKeys'],
+                ['projectId' => $this->projectId] + $options,
+                ['resultLimit' => $resultLimit]
+            )
+        );
+    }
+
+    /**
+     * Lazily instantiate an HMAC Key instance using an Access ID.
+     *
+     * Example:
+     * ```
+     * $hmacKey = $storage->hmacKey($accessId);
+     * ```
+     *
+     * @param string $accessId The ID of the HMAC Key.
+     * @param array $metadata [optional] HMAC key metadata.
+     * @return HmacKey
+     */
+    public function hmacKey($accessId, array $metadata = [])
+    {
+        $this->requireProjectId();
+
+        return new HmacKey($this->connection, $this->projectId, $accessId, $metadata);
+    }
+
+    /**
+     * Creates a new HMAC key for the specified service account.
+     *
+     * Please note that the HMAC secret is only available at creation. Make sure
+     * to note the secret after creation.
+     *
+     * Example:
+     * ```
+     * $hmacKey = $storage->createHmacKey('account@myProject.iam.gserviceaccount.com');
+     * $secret = $hmacKey->secret();
+     * ```
+     *
+     * @param string $serviceAccountEmail Email address of the service account.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $userProject If set, this is the ID of the project which
+     *           will be billed for the request.
+     * }
+     * @return HmacKey
+     */
+    public function createHmacKey($serviceAccountEmail, array $options = [])
+    {
+        $this->requireProjectId();
+
+        $res = $this->connection->createHmacKey([
+            'projectId' => $this->projectId,
+            'serviceAccountEmail' => $serviceAccountEmail
+        ] + $options);
+
+        return new HmacKey(
+            $this->connection,
+            $this->projectId,
+            $res['metadata']['accessId'],
+            $res['metadata'],
+            $res['secret']
```

Comment:
What about returning the `secret` in a separate data structure, not inside the `HmacKeyMetadata` object? It would make it more clear that the secret is only returned on creation, and by not having it part of `HmacKey`.

The [python](https://github.com/googleapis/google-cloud-python/issues/7851) implementation currently does this, by returning a tuple {secret, HmacKeyMetadata} on calls to `createHmacKey`, and I plan to implement the same in Node.js.

Reply:
That sounds very reasonable. For publicly accessible methods, we have tended towards returning a single value, as a tuple isn't really a construct in PHP. To better support typehints for IDEs, WDYT about a `CreateHmacKeyResponse` class which has two getters? `hmacKey` and `secret`?

Reply:
This is done!